submissions/SamVal007/Frogger game task/app.js

```diff
@@ -0,0 +1,129 @@
+const playerStartValue = {
+    "x": 200,
+    "y": 400,
+    "icon": "images/char-boy.png",
+    "stepX": 100,
+    "stepY": 82,
+    "startScore": 0
+};
+
+const enemyStartValue = {
+    "x": 0,
+    "icon": "images/enemy-bug.png",
+    "minX": -50,
+    "maxX": 510,
+    "firstEnemy": 63,
+    "secondEnemy": 147,
+    "thirdEnemy": 230
+};
+
+const coordsGameGround = {
+    "min": 0,
+    "max": 400,
+    "delay": 200,
+    "conflictX": 80,
+    "conflictY": 60,
+    "minSpeed": 100,
+    "maxSpeed": 222
+};
+
+class Character {
+    constructor(x, y, sprite) {
+        this.x = x;
+        this.y = y;
+        this.sprite = sprite;
+    }
+
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+}
+
+class Enemy extends Character {
+    constructor(x, y, sprite, player) {
+        super(x, y, sprite);
+        this.speed = this.getRandomSpeed();
+        this.player = player;
+    }
+
+    update(dt) {
+        this.x += this.speed * dt;
+        if(this.x > enemyStartValue.maxX) {
+            this.x = enemyStartValue.minX;
+            this.speed = this.getRandomSpeed();
+        }
+        this.isCollision();
+    }
+
+    isCollision() {
+        if(player.x < this.x + coordsGameGround.conflictX && player.x + coordsGameGround.conflictX > this.x && player.y < this.y + coordsGameGround.conflictY && coordsGameGround.conflictY + player.y > this.y) {
+            player.x = playerStartValue.x;
+            player.y = playerStartValue.y;
+            this.player.minusScore();
+        }
+    }
```

Comment:
This method refers to a global variable `player`. Classes shouldn't refer to global variables.

Although referring to global constants is ok. In this particular task referring to game engine resources is also OK. However, the global variable `player` is neither const nor engine's entity.

Reply:
As I understand it  -we can use here just "player.minusScore();" without 'this'. Right?

Reply:
I got it! Yeah, really i should put there ` this.player`